### PR TITLE
Spec fixes

### PR DESCRIPTION
**Pull request by jko0531**

### CODE REVIEW

**Comment by jko0531** [comment]
The updated csv files (colleges.csv, college_labels.csv) can be found in the spec-fixes branch in assets folder in okcollege-dev

File: college_preprocessor.py [hunk 1]
```diff
@@ -6,81 +6,180 @@
 import pika
```
Comment (seveibar) [inline college_preprocessor.py:15]: A descriptive comment at the top of the module explaining the general purpose of the module and it's process would be useful here, [here's an example from the survey preprocessor](https://github.com/rpiml/survey-preprocessor/blob/master/src/parse.js)

**Review by seveibar — CHANGES_REQUESTED** [review]

**Comment by seveibar** [comment]
does this fix #4 ?

**Comment by seveibar** [comment]
Other than the descriptive comment, this is looking pretty good.

File: college_preprocessor.py [hunk 2]
```diff
@@ -6,81 +6,180 @@
...
+			return 0
+		if value == 'Less selective':
+			return 1
+		if value == 'Selective':
+			return 2
+		if value == 'More selective':
+			return 3
+		if value == 'Most selective':
+			return 4
+	return 0
```
Comment (jowlee) [inline college_preprocessor.py]: Does not having a value that matches the choices mean that the school is 'least selective' or that it wasn't included. If it wasn't included, we should probably modify the default to be a unique number like -1 or shift the other values up one value.

File: college_preprocessor.py [hunk 3]
```diff
@@ -6,81 +6,180 @@
...
+	if isinstance(value, basestring):
+		if value == 'Neither SAT nor ACT':
+			return 0
+		if value == 'SAT':
+			return 1
+		if value == 'ACT':
+			return 2
+		if value == 'Either SAT or ACT':
+			return 3
+	return 0
```
Comment (jowlee) [inline college_preprocessor.py]: Does not having a value for SAT or ACT imply that the school accepts neither? If not then it should be return a unique number.

File: college_preprocessor.py [hunk 4]
```diff
@@ -89,32 +188,68 @@ def parsecolleges(college_file, columns):
...
+    df = pd.read_csv(college_file, header=None)
     df.columns = columns
-    df['SAT1'] = df['SAT/ACT 25th-75th percentile'].apply(SAT1)
-    df['SAT2'] = df['SAT/ACT 25th-75th percentile'].apply(SAT2)
-    df['ACT1'] = df['SAT/ACT 25th-75th percentile'].apply(ACT1)
-    df['ACT2'] = df['SAT/ACT 25th-75th percentile'].apply(ACT2)
-    df['Student-faculty ratio'] = df['Student-faculty ratio'].apply(ratio)
+
+    # change all 'N/A' values to 0
+    df.fillna(0, inplace=True)
```
Comment (jowlee) [inline college_preprocessor.py:208]: If 0 is the default value for N/A then it should not associated with another value like 'No' or 'least-selective'

File: college_preprocessor.py (hunk 2 above)
Comment (jko0531) [reply college_preprocessor.py]: good point, I've changed all N/A values in categories to be -1.

File: college_preprocessor.py (hunk 4 above)
Comment (jko0531) [reply college_preprocessor.py:208]: After making the changes above, this won't be an issue anymore

**Comment by jko0531** [comment]
Not sure where the vendor folder came from...is that a side effect from the docker-compose?

**Comment by seveibar** [comment]
I also don't know, but you'll have to do an interactive rebase and remove it from the commit before we merge or else our history will get messed up here :/

**Comment by seveibar** [comment]
(after the interactive rebase you'll have to force push btw)

**Comment by jko0531** [comment]
not sure if I did the rebase stuff right...but college_preprocessor.py looks good now, with the vendor stuff removed.

**Comment by seveibar** [comment]
looks right to me

**Review by seveibar — APPROVED** [review]

**Comment by seveibar** [comment]
i looked into it a bit btw and yes the vendor stuff is created by docker-compose, it basically caches the pip stuff, just add vendor to the .gitignore so it's not accidentally committed in the future

**Pull request merged**